pkg/obfuscator/ip_test.go

```diff
@@ -59,10 +61,81 @@ func TestIPObfuscator(t *testing.T) {
 		},
 	} {
 		t.Run(tc.name, func(t *testing.T) {
-			o := NewIPObfuscator()
+			o := NewIPObfuscator(schema.ObfuscateReplacementTypeStatic)
 			output := o.Contents(tc.input)
 			assert.Equal(t, tc.output, output)
 			assert.Equal(t, tc.report, o.ReportingResult())
 		})
 	}
 }
+
+func TestIPObfuscatorConsistent(t *testing.T) {
+	for _, tc := range []struct {
+		name   string
+		input  []string
+		output []string
+		report map[string]string
+	}{
+		{
+			name:   "valid ipv4 address",
+			input:  []string{"received request from 192.168.1.10"},
+			output: []string{"received request from x-ipv4-000001-x"},
+			report: map[string]string{"192.168.1.10": "x-ipv4-000001-x"},
```

Comment:
what's your overflow behaviour for your template pattern "x-ipv4-000001-x"?  maybe add a test that starts at `x-ipv4-999999-x` and check the result? 

probably warrants for a proper error message before the "Sprintf"

Reply:
Since I don't expect there will be more than a million IP addresses to obfuscate I've added a panic when the counter would overflow. We can also not have the panic because if the number exceeds the formatting directive then the formatting directive is ignored.